apis/testing/conditions.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2019 The Knative Authors
+ Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/knative/pkg/apis"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// ConditionCheckable is an interface that most of our status types implement.
+type ConditionCheckable interface {
+	IsReady() bool
+	GetCondition(t apis.ConditionType) *apis.Condition
```

Comment:
🤔  Thinking about this with the refactoring that landed recently, you could just do all of this in terms of `duckv1beta1.Status`.  I think this is fine either way, but what you choose to do with that information is up to you :)

Reply:
Yeah, so the reason I didn't use duck, is because I wanted IsReady :-)

Reply:
Readid with the duck, since I removed IsReady.